### PR TITLE
Feat: add archived channel and category sorting feature

### PR DESCRIPTION
**Pull request by Tuan2002**
## Summary

Implements two clan settings pages that previously showed "Coming soon" — **Category Order** and **Archived Channels** — with full API/store wiring aligned to the mezon web app. Also hides always-visible scrollbars on the clan channel list and DM chat list sidebars for a cleaner layout.

### Category Order

- New **Category Order** settings page at `/chat/clans/{id}/settings/category-order`.
- Lists all reorderable categories for the active clan (excludes favorites and invalid entries).
- Drag-and-drop reorder with a 6-dot grip handle and rounded card rows.
- **Real-time reorder** while dragging; other rows dim during drag.
- Optimistic store update with clan refresh on success; reverts local order on API failure.

### Archived Channels

- New **Archived Channels** settings page at `/chat/clans/{id}/settings/archived-channels`.
- Requires `manageClan` permission (existing sidebar gating).
- Loads archived top-level channels via `ListArchivedChannelDescs` when the page opens.
- Empty state when there are no archived channels.
- Each row shows channel icon (public/private), name, last-activity subtitle, and a **Restore** button.
- Restore calls `ActiveArchivedThread`, removes the row from the list, refreshes clan channels, and shows a success toast.

### Sidebar UX

- Hide always-visible scrollbar thumbs on the **clan channel list** sidebar.
- Hide always-visible scrollbar thumbs on the **DM chat list** sidebar.

### Category Order
- [ ] Open **Clan Settings → Category Order** for a clan with multiple categories.
- [ ] Drag a category up/down; confirm real-time reorder and vertical-only preview (no horizontal drift).
- [ ] Release drag; confirm order persists after leaving settings and returning (auto-save).
- [ ] Reload app / refetch clan; confirm sidebar category order matches.
- [ ] Simulate API failure (offline); confirm list reverts without toast spam.
- [ ] Confirm favorites category is not listed.

### Archived Channels
- [ ] Open **Clan Settings → Archived Channels** as a user with `manageClan`.
- [ ] With no archived channels, confirm empty state.
- [ ] Archive a channel from elsewhere (web or future desktop menu); confirm it appears in the list.
- [ ] Click **Restore**; confirm row disappears, channel reappears in sidebar, success toast shown.
- [ ] Confirm page is inaccessible without `manageClan` (redirect to allowed page).

### Scrollbars
- [ ] Clan channel list: scroll with wheel/trackpad; confirm no visible scrollbar thumb.
- [ ] DM list: same check.

#### Category Order
Screenshot 2026-07-19 at 00 11 39

#### Archived Channels
Screenshot 2026-07-19 at 00 12 22

### CODE REVIEW

File: crates/mezon-store/src/channel.rs [hunk 1]
```diff
@@ -1124,6 +1174,50 @@ impl ChannelList {
...
+                    c.order
+                }
+            });
+            cx.notify();
+        }
+        let api = self.api.clone();
+        cx.spawn(async move |this, cx| {
+            api.update_category_order(clan_id.get(), &payload)
+                .await
+                .map_err(|e| e.to_string())?;
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: This ? returns before refresh_clan, so the optimistic cache mutation above is never reverted. Settings page reverts its own list but the sidebar keeps the rejected order. Call refresh_clan on the error path too.

File: crates/mezon-ui/src/clan/settings/archived_channel_page.rs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                            .map(|desc| ArchivedChannelRow {
+                                channel_id: desc.channel_id,
+                                channel_label: desc.channel_label.into(),
+                                channel_private: desc.channel_private,
+                                last_active_timestamp: desc.last_active_timestamp,
+                            })
+                            .collect();
+                    }
+                    Err(err) => {
+                        tracing::error!("fetch archived channels failed: {err}");
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/archived_channel_page.rs:91]: On fetch failure loading is cleared and channels stays empty, so the user sees the no-archived-channels empty state instead of an error. Add a distinct error state with retry.

File: crates/mezon-ui/src/clan/settings/archived_channel_page.rs [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+            let _ = this.update(cx, |this, cx| {
+                this.restoring = None;
+                match result {
+                    Ok(()) => {
+                        this.channels.retain(|row| row.channel_id != channel_id);
+                        this.channel_list
+                            .update(cx, |store, cx| store.refresh_clan(clan_id, cx));
+                    }
+                    Err(err) => {
+                        tracing::error!("restore archived channel failed: {err}");
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/archived_channel_page.rs]: Restore failure is log-only while success shows a toast. Shell has an error variant, use it.

File: crates/mezon-ui/src/clan/settings/category_sort_page.rs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+            Ok(()) => {
+                let _ = this.update(cx, |this, cx| {
+                    this.saved_categories = this.categories.clone();
+                    this.has_changed = false;
+                    this.saving = false;
+                    cx.notify();
+                });
+            }
+            Err(err) => {
+                tracing::error!("update categories order failed: {err}");
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/category_sort_page.rs:213]: Save failure is log-only, the list just snaps back with no explanation. CategoryOrder is visible to every member (visible_in_sidebar returns true), so a server rejection is a common path here.

File: crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs [hunk 5]
```diff
@@ -855,13 +855,7 @@ impl Render for ChannelSidebar {
...
                     .child(list_element)
                     .children(skeleton_overlay)
-                    .children(mention_button)
-                    .custom_scrollbars(
-                        Scrollbars::always_visible(ScrollAxes::Vertical)
-                            .tracked_scroll_handle(&self.list_state),
-                        window,
-                        cx,
-                    ),
+                    .children(mention_button),
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/channel_sidebar/mod.rs]: Dropping custom_scrollbars removes the scrollbar entirely, not just the always-visible thumb: no scroll position indicator and no drag-to-scroll. ShowScrollbar defaults to Auto, so Scrollbars::new(ScrollAxes::Vertical) gives the autohide behaviour the description asks for.

File: crates/mezon-ui/src/sidebar/direct_sidebar.rs [hunk 6]
```diff
@@ -486,19 +485,7 @@ impl Render for DirectSidebar {
...
-                    .relative()
-                    .child(list)
-                    .custom_scrollbars(
-                        Scrollbars::always_visible(ScrollAxes::Vertical)
-                            .tracked_scroll_handle(&self.list_scroll),
-                        window,
-                        cx,
-                    ),
-            )
+            .child(div().flex_1().min_h_0().relative().child(list))
```
Comment (huylexuan44) [inline crates/mezon-ui/src/sidebar/direct_sidebar.rs]: Same as the clan sidebar: use Scrollbars::new instead of removing custom_scrollbars, so the thumb autohides but stays draggable.

File: crates/mezon-ui/src/clan/settings/category_sort_page.rs [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+        });
+
+        row
+    }
+}
+
+impl Render for CategorySortPage {
+    fn render(&mut self, _window: &mut Window, cx: &mut Context<Self>) -> impl IntoElement {
+        if self.dragging_index.is_some() && !cx.has_active_drag() {
+            self.finish_drag(cx);
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/category_sort_page.rs]: Mutating state and kicking off an API write from inside render. This is the cancelled-drag fallback, so wrap it in cx.defer to move the work out of the render pass.

File: crates/mezon-ui/src/clan/settings/category_sort_page.rs [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+            saving: false,
+            dragging_index: None,
+            drag_pointer_y: None,
+            _channel_sub: cx.observe(&channel_list, |this, _, cx| {
+                this.resync_from_store(cx);
+            }),
+        }
+    }
+
+    pub fn release(&mut self) {}
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/category_sort_page.rs]: Empty no-op but release_page still calls it. Either delete it and the match arm, or use it to cancel the detached save task.

File: crates/mezon-ui/src/clan/settings/category_sort_page.rs [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+                tracing::error!("update categories order failed: {err}");
+                let _ = this.update(cx, |this, cx| {
+                    this.categories = this.saved_categories.clone();
+                    this.has_changed = false;
+                    this.saving = false;
+                    cx.notify();
+                });
+            }
+        })
+        .detach();
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/category_sort_page.rs]: ArchivedChannelPage stores its tasks in _fetch_task/_restore_task fields. Store this one too instead of detaching.

File: crates/mezon-ui/src/clan/settings/category_sort_page.rs [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+            .iter()
+            .filter(|category| Self::is_valid_sort_category(&category.id))
+            .collect();
+        categories.sort_by_key(|category| category.order);
+        categories
+            .into_iter()
+            .map(|category| CategorySortItem {
+                id: category.id.clone(),
+                name: category.name.clone(),
+                category_id: category.id.parse().unwrap_or(0),
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/category_sort_page.rs]: Dead fallback: is_valid_sort_category already guarantees the parse succeeds and is > 0.

File: crates/mezon-ui/src/clan/settings/archived_channel_page.rs [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+            .bg(theme.tokens.bg_item_theme_hover)
+            .shadow_sm()
+            .child(
+                div()
+                    .flex_shrink_0()
+                    .size(px(32.0))
+                    .flex()
+                    .items_center()
+                    .justify_center()
+                    // .rounded_md()
```
Comment (huylexuan44) [inline crates/mezon-ui/src/clan/settings/archived_channel_page.rs]: Commented-out styling, remove or restore.

**Comment by huylexuan44** [comment]
Review summary — solid work. The ClanSettingsPage enum variants and permission gating already existed and are used correctly, all 6 clanSettings.archivedChannels / common.justNow i18n keys resolve in assets/i18n/en.json, the three MezonTransport methods already exist, and CategorySortPage correctly stores its observer in a Subscription field instead of detaching.

Needs a rebase first: the branch is 35 commits behind develop and 7 of the 9 changed files conflict (app_api.rs, transport_runtime.rs, channel.rs, clan_setting_screen.rs, settings/mod.rs, channel_sidebar/mod.rs, direct_sidebar.rs). Note channel_sidebar/mod.rs and app_api.rs are also touched by open PRs #150 and #151, so whoever rebases last pays the most — this one is the oldest, so it is probably cheapest to land first.

Two things to fix beyond that:

- update_categories_order mutates the store cache optimistically but never reverts it on API failure, so a rejected reorder leaves the sidebar showing the wrong order while the settings page shows the right one.
- All three failure paths are log-only. The worst is the archived fetch, which renders the no-archived-channels empty state after a network error, so the user cannot tell an error from an empty list.

The scrollbar change is also unrelated to archived channels / category sorting and is one of the conflicting hunks — worth splitting out either way.

I did not run a build; every referenced API resolves and nothing looked compile-suspicious.
